storage/kubernetes/types.go

```diff
@@ -10,145 +10,204 @@ import (
 	"github.com/dexidp/dex/storage/kubernetes/k8sapi"
 )
 
+const (
+	// Kinds
+	kindAuthCode        = "AuthCode"
+	kindAuthRequest     = "AuthRequest"
+	kindClient          = "OAuth2Client"
+	kindRefreshToken    = "RefreshToken"
+	kindKeys            = "SigningKey"
+	kindPassword        = "Password"
+	kindOfflineSessions = "OfflineSessions"
+	kindConnector       = "Connector"
+
+	// Resources
+	resourceAuthCode        = "authcodes"
+	resourceAuthRequest     = "authrequests"
+	resourceClient          = "oauth2clients"
+	resourceRefreshToken    = "refreshtokens"
+	resourceKeys            = "signingkeies" // Kubernetes attempts to pluralize.
+	resourcePassword        = "passwords"
+	resourceOfflineSessions = "offlinesessionses" // Again attempts to pluralize.
+	resourceConnector       = "connectors"
+
+	// Singular Resources for CRD references
+	resourceAuthCodeSingular        = "authcode"
+	resourceAuthRequestSingular     = "authrequest"
+	resourceClientSingular          = "oauth2client"
+	resourceRefreshTokenSingular    = "refreshtoken"
+	resourceKeysSingular            = "signingkey"
+	resourcePasswordSingular        = "password"
+	resourceOfflineSessionsSingular = "offlinesessions"
+	resourceConnectorSingular       = "connector"
+
+	// APIs
+	customResourceDefinitionAPI       = "apiextensions.k8s.io/v1beta1"
+	customResourceDefinitionNamespace = ""
+	customResourceDefinitionResource  = "customresourcedefinitions"
+
+	// Dex k8s-specific API
+	apiGroup    = "dex.coreos.com"
+	apiFullPath = apiGroup + "/v1"
```

Comment:
This should be called `apiVersion`. This is a Kubernetes term for the combination of the api group and the version, while I admit it doesn't make as much sense, it's idiomatic so I think we should call it `apiVersion` (plus it's called this throughout the codebase as well)